services/app-api/src/testHelpers/gqlHelpers.ts

```diff
@@ -83,6 +84,9 @@ const constructTestPostgresServer = async (opts?: {
 
     const prismaClient = await sharedTestPrismaClient()
     const postgresStore = opts?.store || NewPostgresStore(prismaClient)
+
+    await insertUserToLocalAurora(postgresStore, context.user)
```

Comment:
what happens if the user already exists? will this silently fail? We should check if there is an error an throw it if it’s not because the user already exists.

Reply:
it looks like he does `lookupUserAurora()` inside `insertUserToLocalAurora()`, but if we want to do it the way that we're currently doing it we'd want to do the checking before calling the insert func. 

It's a good question though, do we prefer doing the insert or should we call [`upsert`](https://www.prisma.io/docs/concepts/components/prisma-client/crud#update-or-create-records), which will just update the entry if it exists and don't do the check at all?

Reply:
> what happens if the user already exists? will this silently fail? We should check if there is an error and throw it if it’s not because the user already exists.

It would silently fail to store the user and return the `context.user`. The function is used in local deployment, mimicking what we do in other environments. 

Your question brings up something I overlooked: if `lookupUserAurora` or storing the user results in an error, it would also cause question tests to fail. The larger issue, I think, is that failing to store a user in any environment will also cause `createQuestions` to fail because it will not be able to relate a question to a user in the database.

@macrael @mojotalantikite @rswerve @haworku Maybe we should discuss this in an informal meeting?

Reply:
> It's a good question though, do we prefer doing the insert or should we call [upsert](https://www.prisma.io/docs/concepts/components/prisma-client/crud#update-or-create-records), which will just update the entry if it exists and don't do the check at all?

Also a good topic to hit on in the informal meeting.

Reply:
Let’s bring this to that meeting, but it doesn’t need to block merging this PR since the current system works. Also it’s totally fine for that insert to fail the tests, anything breaking during a test should fail the test.